WordPress/wordpress-develop
# Try to add extra screen after login asking to confirm admin email

## What is it?

This is a PR with a demo of how the "Is your/this admin email still correct" Trac ticket could function. 
Reference Trac Ticket ID: 46349 ([link](https://core.trac.wordpress.org/ticket/46349))

## How it works
- A new filter function `wp_confirm_admin_email` hooks into `login_redirect`
  - and returns the normal redirect, unless
  - the email has not been confirmed in `ADMIN_EMAIL_MAX_AGE` time
    - The login process is redirected to `action=confirm_admin_email`
    - Which checks for `manage_options`
      - redirects to normal redirect if user doesn't have the cap. or
      - displays the admin email confirm form, otherwise
	- The admin email confirm form, in `wp-login.php`
	  - postpones the confirm, with a remind me later link
	    - which sets an `option` named `admin_email_lifespan`, also
	  - submits a confirmed email address
	    - which is updated using the same `update_option_new_admin_email` function used in the options screen
		- sets an `option` named `admin_email_lifespan`
  - The user end up redirected to the original intented redirect

## How to test

Switch to the branch and update in `default-constants.php` the `ADMIN_EMAIL_MAX_AGE` to a number that makes sense for testing. Make sure your local WP is able to send emails. Login and you should see the new screen.

---

![admin-email-double-check](https://user-images.githubusercontent.com/107534/55352057-a5997200-54c8-11e9-8a31-28f38803ad0a.gif)

---
